indra/sources/signor.py

```diff
@@ -194,24 +216,61 @@ def __init__(self, signor_csv=None, delimiter='\t'):
 
     @staticmethod
     def _get_agent(ent_name, ent_type, id, database):
-        gnd_type = _type_db_map[(ent_type, database)]
-        if gnd_type == 'UP':
-            up_id = id
-            db_refs = {'UP': up_id}
-            name = uniprot_client.get_gene_name(up_id)
-            hgnc_id = hgnc_client.get_hgnc_id(name)
-            if hgnc_id:
-                db_refs['HGNC'] = hgnc_id
-        # Other possible groundings are PUBCHEM and SIGNOR
-        elif gnd_type is not None:
-            assert database in ('PUBCHEM', 'SIGNOR')
-            db_refs = {database: id}
-            name = ent_name
-        # If no grounding, include as an untyped/ungrounded node
+        # Returns a list of agents corresponding to this id
+        # (If it is a signor complex, returns Agent objects corresponding to
+        # each complex constituent)
+        if database == 'SIGNOR' and id in complex_map:
```

Comment:
One additional thing to do for agents involving complexes is to (perhaps in a separate step) create a Complex Statement involving all of the members.